src/components/vr_cooperation/include/vr_cooperation/vr_proxy.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2016, Ford Motor Company
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * Neither the name of the Ford Motor Company nor the names of its contributors
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef SRC_COMPONENTS_VR_COOPERATION_INCLUDE_VR_COOPERATION_VR_PROXY_H_
+#define SRC_COMPONENTS_VR_COOPERATION_INCLUDE_VR_COOPERATION_VR_PROXY_H_
+
+#include "utils/macro.h"
+
+namespace rsdl_api {
+class ProcessDataRequest;
+class ProcessDataResponse;
+}
+
+namespace vr_cooperation {
+class GPBDataSenderReceiver;
+
+class VRProxy {
+ public:
+  /**
+   * @brief Constructor
+   */
+  VRProxy();
+
+  /**
+   * @brief Destructor
+   */
+  ~VRProxy();
+
+  /**
+   * @brief Handle when receive SDLRPC message
+   * @param request_params request parameters
+   */
+  void OnReceived(const rsdl_api::ProcessDataRequest& request_params);
+
+  /**
+   * @brief Post SDLRPC message to queue
+   * @param response_params SDLRPC response message
+   */
+  bool SendMessage(const rsdl_api::ProcessDataResponse& response_params);
```

Comment:
The name of this class depends of how it will be called in proto file. As I see now in protocol which was provided by Elena Saenko its name is ServiceMessage. ProcessDataResponse class will be obtained as result of ServiceMessage params deserialization.